src/Findologic/Config/FindologicConfigService.php

```diff
@@ -241,11 +262,22 @@ private function load(string $salesChannelId, string $languageId): array
     private function buildConfig(FinSearchConfigCollection $configs): array
     {
         $findologicConfig = [];
+        // Set the configuration schema for enabling inheritance
+        foreach (self::CONFIG_KEYS as $configKey) {
+            $findologicConfig[$configKey] = null;
+        }
+
         foreach ($configs as $config) {
-            $keyExists = array_key_exists($config->getConfigurationKey(), $findologicConfig);
-            if (!$keyExists || !Utils::isEmpty($config->getConfigurationValue())) {
-                $findologicConfig[$config->getConfigurationKey()] = $config->getConfigurationValue();
+            $value = $config->getConfigurationValue();
+            $key = $config->getConfigurationKey();
+            $inheritedValuePresent = array_key_exists($key, $findologicConfig);
+            $isValueEmpty = !is_bool($value) && Utils::isEmpty($value);
```

Comment:
But a boolean false is not technically an empty value. So we might introduce some issues.

Reply:
Ah I see, I misinterpreted this check. Why is a boolean empty in this case then? It isn't very obvious.

Reply:
Yeah I already fixed it in the last update.

Reply:
What do you mean by that? The check `is_bool` doesn't even need to be here. Why was this check even introduced and is it required? Here I have a few examples:

`$value = true` => `!is_bool($value) is false` `Utils::isEmpty($value) is false` => `$isValueEmpty is false`
`$value = true` => `!is_bool($value) is false` `Utils::isEmpty($value) is false` => `$isValueEmpty is false`
`$value = "test"` => `!is_bool($value) is true` `Utils::isEmpty($value) is false` => `$isValueEmpty is false`
`$value = ["test"]` => `!is_bool($value) is true` `Utils::isEmpty($value) is false` => `$isValueEmpty is false`

The check `!is_bool` seems to me as it is not required in any case. I could be wrong but I am curious why it has been introduced in the first place, since it seems like it has no real impact on the returned result.

Please remove it if it turns out to really have no impact.

Reply:
Did you check my previous comment? I have already removed that check. It was how shopware was doing it so I figured it might have some significance but then in our case we did not have any importance so I have removed that check already.